MSVC with Ninja fails to compile SharedHandles sample

When trying to compile the `SharedHandle` sample with MSVC (2022) using `Ninja` (-G Ninja), it is not able to construct a shared handle of `vk::SurfaceKHR`:

https://github.com/KhronosGroup/Vulkan-Hpp/blob/0a7333161db0821452ad950760e48abc35ea0fa3/samples/SharedHandles/SharedHandles.cpp#L89

fails with the following output:

```bash
FAILED: [code=2] samples/SharedHandles/CMakeFiles/SharedHandles.dir/SharedHandles.cpp.obj 
C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1444~1.352\bin\Hostx86\x86\cl.exe  /nologo /TP -DVK_NO_PROTOTYPES -DVK_USE_PLATFORM_WIN32_KHR -DVULKAN_HPP_DISPATCH_LOADER_DYNAMIC -DVULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1 -ID:\a\Vulkan-Hpp\Vulkan-Hpp -ID:\a\Vulkan-Hpp\Vulkan-Hpp\Vulkan-Headers\include -ID:\a\Vulkan-Hpp\Vulkan-Hpp\glm -ID:\a\Vulkan-Hpp\Vulkan-Hpp\glfw\include -ID:\a\Vulkan-Hpp\Vulkan-Hpp\glslang\glslang\.. -ID:\a\Vulkan-Hpp\Vulkan-Hpp\build\include -ID:\a\Vulkan-Hpp\Vulkan-Hpp\glslang\SPIRV\.. /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /WX /showIncludes /Fosamples\SharedHandles\CMakeFiles\SharedHandles.dir\SharedHandles.cpp.obj /Fdsamples\SharedHandles\CMakeFiles\SharedHandles.dir\ /FS -c D:\a\Vulkan-Hpp\Vulkan-Hpp\samples\SharedHandles\SharedHandles.cpp
D:\a\Vulkan-Hpp\Vulkan-Hpp\samples\SharedHandles\SharedHandles.cpp(89): error C2440: 'initializing': cannot convert from 'initializer list' to 'vk::SharedHandle<vk::SurfaceKHR>'
D:\a\Vulkan-Hpp\Vulkan-Hpp\samples\SharedHandles\SharedHandles.cpp(89): note: 'vk::SharedHandle<vk::SurfaceKHR>::SharedHandle': function does not take 2 arguments
D:\a\Vulkan-Hpp\Vulkan-Hpp\vulkan/vulkan_shared.hpp(304): note: could be 'vk::SharedHandle<vk::SurfaceKHR>::SharedHandle(HandleType,vk::detail::ObjectDestroyShared<HandleType>) noexcept'
        with
        [
            HandleType=vk::SurfaceKHR
        ]
D:\a\Vulkan-Hpp\Vulkan-Hpp\vulkan/vulkan_shared.hpp(295): note: or       'vk::SharedHandle<vk::SurfaceKHR>::SharedHandle(HandleType,vk::SharedHandle<vk::Instance>,vk::SharedHandle<vk::NoDestructor>,const Dispatcher &) noexcept'
        with
        [
            HandleType=vk::SurfaceKHR
        ]
D:\a\Vulkan-Hpp\Vulkan-Hpp\vulkan/vulkan_shared.hpp(285): note: or       'vk::SharedHandle<vk::SurfaceKHR>::SharedHandle(HandleType,vk::SharedHandle<vk::Instance>,vk::detail::ObjectDestroyShared<HandleType>) noexcept'
        with
        [
            HandleType=vk::SurfaceKHR
        ]
D:\a\Vulkan-Hpp\Vulkan-Hpp\samples\SharedHandles\SharedHandles.cpp(89): note: while trying to match the argument list '(VkSurfaceKHR, vk::SharedInstance)'
```

--- a/samples/SharedHandles/SharedHandles.cpp
+++ b/samples/SharedHandles/SharedHandles.cpp
@@ -86,7 +86,7 @@ public:
     VkResult     err = glfwCreateWindowSurface( instance.get(), window.handle, nullptr, &surface );
     if ( err != VK_SUCCESS )
       throw std::runtime_error( "Failed to create window!" );
-    vk::SharedSurfaceKHR sharedSurface{ surface, instance };
+    vk::SharedSurfaceKHR sharedSurface{ static_cast<vk::SurfaceKHR>( surface ), instance };
 
     auto graphicsAndPresentQueueFamilyIndex = vk::su::findGraphicsAndPresentQueueFamilyIndex( physicalDevice, sharedSurface.get() );
     device = vk::SharedDevice{ vk::su::createDevice( physicalDevice, graphicsAndPresentQueueFamilyIndex.first, vk::su::getDeviceExtensions() ) };
@@ -109,16 +109,12 @@ public:
     std::transform( swapChainData.images.begin(),
                     swapChainData.images.end(),
                     std::back_inserter( images ),
-                    [this]( vk::Image image ) {
-                      return vk::SharedImage{ image, device, vk::SwapchainOwns::yes };
-                    } );
+                    [this]( vk::Image image ) { return vk::SharedImage{ image, device, vk::SwapchainOwns::yes }; } );
 
     std::transform( swapChainData.imageViews.begin(),
                     swapChainData.imageViews.end(),
                     std::back_inserter( imageViews ),
-                    [this]( vk::ImageView imageView ) {
-                      return vk::SharedImageView{ imageView, device };
-                    } );
+                    [this]( vk::ImageView imageView ) { return vk::SharedImageView{ imageView, device }; } );
     commandPool =
       vk::SharedCommandPool{ device->createCommandPool( { vk::CommandPoolCreateFlagBits::eResetCommandBuffer, graphicsAndPresentQueueFamilyIndex.first } ),
                              device };
@@ -230,9 +226,9 @@ public:
     vk::Result result = presentQueue->presentKHR( vk::PresentInfoKHR( {}, swap, currentBuffer ) );
     switch ( result )
     {
-      case vk::Result::eSuccess: break;
+      case vk::Result::eSuccess      : break;
       case vk::Result::eSuboptimalKHR: std::cout << "vk::Queue::presentKHR returned vk::Result::eSuboptimalKHR !\n"; break;
-      default: assert( false );  // an unexpected result is returned !
+      default                        : assert( false );  // an unexpected result is returned !
     }
     std::this_thread::sleep_for( std::chrono::milliseconds( 1000 ) );
 

--- a/tests/UniqueHandleDefaultArguments/UniqueHandleDefaultArguments.cpp
+++ b/tests/UniqueHandleDefaultArguments/UniqueHandleDefaultArguments.cpp
@@ -17,23 +17,25 @@
 // Should be used on 64 bit only, as on 32 bit the test is ambiguous.
 
 #ifdef VULKAN_HPP_USE_CXX_MODULE
-  #include <vulkan/vulkan.h>
-  import vulkan_hpp;
+#  include <vulkan/vulkan.h>
+import vulkan_hpp;
 #else
-# include "vulkan/vulkan.hpp"
+#  include "vulkan/vulkan.hpp"
 #endif
 
 #if VULKAN_HPP_DISPATCH_LOADER_DYNAMIC == 1
-namespace vk {
-  namespace detail {
+namespace vk
+{
+  namespace detail
+  {
     DispatchLoaderDynamic defaultDispatchLoaderDynamic;
-  }
-}
+  }  // namespace detail
+}  // namespace vk
 #endif
 
 int main( int /*argc*/, char ** /*argv*/ )
 {
   VkSurfaceKHR surface       = 0;
-  auto         uniqueSurface = vk::UniqueSurfaceKHR( surface, vk::Instance() );
+  auto         uniqueSurface = vk::UniqueSurfaceKHR( static_cast<vk::SurfaceKHR>( surface ), vk::Instance() );
   return 0;
 }

